deploy/coins/bitcoin/Dockerfile

```diff
@@ -3,34 +3,51 @@ FROM alpine:latest as builder
 
 ENV BITCOIN_VERSION=23.0
 ENV GLIBC_VERSION=2.28-r0
+ENV SIG_REQ=3
 ENV BITCOIN_DATA=/home/bitcoin/.bitcoin
 
 WORKDIR /home/bitcoin
 
+
 RUN apk update \
 	&& apk --no-cache add ca-certificates gnupg bash su-exec 
 
+SHELL ["/bin/bash", "-c"]
 # Get Binary
 RUN wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/bitcoin-${BITCOIN_VERSION}-x86_64-linux-gnu.tar.gz \
 	&& wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/SHA256SUMS \
 	&& wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/SHA256SUMS.asc
 
 # Verify Binary
 # For the PGP keys of builders and developers, refer to 
-# https://github.com/bitcoin/bitcoin/tree/master/contrib/builder-keys 
-ENV KEYS 152812300785C96444D3334D17565732E08E5E41 0AD83877C1F0CD1EE9BD660AD7CC770B81FD22A8 590B7292695AFFA5B672CBB2E13FC145CD3F4304 28F5900B1BB5D1A4B6B6D1A9ED357015286A333D 637DB1E23370F84AFF88CCE03152347D07DA627C CFB16E21C950F67FA95E558F2EEB9F5CC09526C1 F4FC70F07310028424EFC20A8E4256593F177720 D1DBF2C4B96F2DEBF4C16654410108112E7EA81F 287AE4CA1187C68C08B49CB2D11BD4F33F1DB499 F9A8737BF4FF5C89C903DF31DD78544CF91B1514 9DEAE0DC7063249FB05474681E4AED62986CD25D E463A93F5F3117EEDE6C7316BD02942421F4889F 9D3CC86A72F8494342EA5FD10A41BDC3F4FAFF1C 4DAF18FE948E7A965B30F9457E296D555E7F63A7 28E72909F1717FE9607754F8A7BEB2621678D37D 74E2DEF5D77260B98BC19438099BAD163C70FBFA
-ENV KEYS2 637DB1E23370F84AFF88CCE03152347D07DA627C 82921A4B88FD454B7EB8CE3C796C4109063D4EAF
+# https://github.com/bitcoin/bitcoin/tree/master/contrib/builder-keys
+# Serai recognizes a handselected reduced set of keys from the list, prefixes -- 
+# marco -- B8B3
+# laanwj -- 71A3
+# fanquake -- E777
+# achow101 -- 1528
+# glozow -- 6B00
+# hebasto -- D1DB
+ENV KEYS  B8B3F1C0E58C15DB6A81D30C3648A882F4316B9B 71A3B16735405025D447E8F274810B012346C9A6 E777299FC265DD04793070EB944D35F9AC3DB76A
+ENV KEYS2 152812300785C96444D3334D17565732E08E5E41 6B002C6EA3F91B1B0DF0C9BC8F617F1200A6D25C D1DBF2C4B96F2DEBF4C16654410108112E7EA81F
 
+# Use footprints to get keys from servers. 2 Different servers used.
 RUN gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys ${KEYS} \
-	&& gpg --keyserver hkp://keys.openpgp.org:80 --recv-keys ${KEYS2} \
-	# removing problematic sigs from SHA256SUMS.asc (using sed)
-	&& sed -i -e '156,162d' SHA256SUMS.asc \ 
-	&& gpg --verify SHA256SUMS.asc SHA256SUMS \
-	&& grep bitcoin-${BITCOIN_VERSION}-x86_64-linux-gnu.tar.gz SHA256SUMS | sha256sum -c
+	&& gpg --keyserver hkp://keys.openpgp.org:80 --recv-keys ${KEYS2}
+
+# Capture error output and count good signature occurrence
+# if above threshold continue
+# else exit code 2
+RUN { err="$( { gpg --verify --status-fd 1 --verify SHA256SUMS.asc SHA256SUMS; } 2>&1 >&3 3>&- )"; } 3>&1; num="$(echo $err | grep -o 'Good signature' | wc -l)" \
```

Comment:
This is vulnerable to keys literally named "Good signature". It is also vulnerable to signature replay to bypass the threshold with just a single party.

`awk '/-----BEGIN PGP SIGNATURE-----/{n++}{print >"sig" n ".txt" }' SHA256SUMS.asc`